kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -223,6 +233,46 @@ def _create_kubelet_prometheus_instance(self, instance):
         )
         return kubelet_instance
 
+    def _create_pod_tags_by_pvc(self, pods):
+        """
+        Return a map, e.g.
+            {
+                "<kube_namespace>/<persistentvolumeclaim>": [<list_of_pod_tags>],
+                "<kube_namespace1>/<persistentvolumeclaim1>": [<list_of_pod_tags1>],
+            }
+        that can be used to add pod tags to associated volume metrics
+        """
+        pod_tags_by_pvc = defaultdict(set)
+        for pod in pods['items']:
+            # get kubernetes namespace of pvc
+            kube_ns = pod.get('metadata', {}).get('namespace')
+            if not kube_ns:
+                continue
+
+            # get pod id and tags from tagger
+            pod_id = pod.get('metadata', {}).get('uid')
+            if not pod_id:
+                self.log.debug('skipping pod with no uid')
+                continue
+            tags = tagger.tag('kubernetes_pod_uid://%s' % pod_id, tagger.LOW) or None
+            if not tags:
+                continue
+
+            # remove `persistentvolumeclaim` tags because there can be more than one PVC on a pod
+            # the relevant PVC tag will be added later from the label
+            tags = [t for t in tags if not t.startswith('persistentvolumeclaim:')]
+
+            # get persistentvolumeclaim
+            volumes = pod.get('spec', {}).get('volumes')
+            if not volumes:
+                continue
```

Comment:
can we move this up before calling the tagger? to avoid calling the tagger and filter the tags if the pod doesn't have volumes